ipld/encoding/src/ipld_block.rs

```diff
@@ -1,16 +1,30 @@
+use std::fmt::{Debug, Formatter};
+
 // Copyright 2021-2023 Protocol Labs
 // SPDX-License-Identifier: Apache-2.0, MIT
 use serde::de::value;
 use {serde, serde_ipld_dagcbor};
 
 use crate::{CodecProtocol, Error, RawBytes, DAG_CBOR, IPLD_RAW};
 
-#[derive(Debug, PartialEq, Eq, Clone, Default)]
+#[derive(PartialEq, Eq, Clone, Default)]
 pub struct IpldBlock {
     pub codec: u64,
     pub data: Vec<u8>,
 }
 
+impl Debug for IpldBlock {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "IpldBlock {{ ")?;
```

Comment:
Use https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.debug_struct.

Reply:
originally avoided because you cant hex encode in a loop like that without making a buffer which i didnt want. Didn't think hard enough about it though, new solution that doesnt need a container and still uses debug struct :)